### PR TITLE
LITE-20722 Adding the new methods to the template. Adding schedulable…

### PR DESCRIPTION
**Pull request by vlas**
LITE-20722 Adding the new methods to the template. Adding schedulables section to extension.json and HOWTO.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.json [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
-    "changelog_url": "https://example.com/CHANGELOG.md",
-    "capabilities": {
+  "name": "{{ cookiecutter.project_name }}",
+  "description": "{{ cookiecutter.description }}",
+  "version": "{{ cookiecutter.version }}",
+  "readme_url": "https://example.com/README.md",
+  "changelog_url": "https://example.com/CHANGELOG.md",
+  "capabilities": {
+  },
+  "schedulables": [
```
Comment (marcserrat) [inline {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.json]: I don't understand why by default we want to inject this mock scheduled and this vars.....
i believe this must not be here at all......
You have 2 options:

a) You ask if you want SAMPLES about this and then you create it
b) You put it on the readme files as explanation and link to documentation.

Comment (vlas) [reply {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.json]: Thanks, I will then put it into readme only.

Comment (vlas) [reply {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.json]: this has been adjusted and in addition to readme, also we are asking the user if it needs an example in extension.json/py for schedulables and variables.

**Review by marcserrat — APPROVED** [review]

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -22,6 +22,42 @@
...
+    'uploaded',
+    'invalid',
+    'processing',
+    'processed',
+    'ready',
+    'rejected',
+    'pending',
+    'accepted',
+    'closed',
+    'deleted',
```
Comment (ghost) [inline hooks/post_gen_project.py]: Deleted is not supported

**Review by ghost — CHANGES_REQUESTED** [review]

File: hooks/post_gen_project.py (hunk 2 above)
Comment (vlas) [reply hooks/post_gen_project.py]: removed, thanks

File: {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py [hunk 3]
```diff
@@ -138,4 +138,63 @@ class {{ cookiecutter.project_name|title|replace(" ", "") }}Extension(Extension)
...
+        return ProcessingResponse.done()
+
+    {% endif -%}
+
+    {% if cookiecutter.include_schedules_example == 'y' -%}
+    {% if cookiecutter.use_asyncio == 'y' %}async {% endif %}def execute_scheduled_processing(self, schedule):  # pragma: no cover
+        self.logger.info(
+            f"Received event for schedule  {schedule['id']}",
+        )
+        return ProcessingResponse.done()
```
Comment (ghost) [inline {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py]: In the case of schedulable methods, the response must be ScheduledExecutionResponse.done()

Comment (ghost) [reply {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py]: You also have to import ScheduledExecutionResponse

**Review by ghost — CHANGES_REQUESTED** [review]
You have to modify the docker-compose.yml to fix the image version (24.3) and the pyproject.toml to have the 24.* runner as a dependency.

File: {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py (hunk 3 above)
Comment (vlas) [reply {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py]: thanks, fixed this

**Review by ghost — APPROVED** [review]

**Review by gab832 — APPROVED** [review]
Great!

File: {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py (hunk 3 above)
Comment (vlas) [reply {{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/extension.py]: modified also docker-compose.yml

**Review by marcserrat — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**